mlflow/models/evaluation/default_evaluator.py

```diff
@@ -332,6 +347,95 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _load_custom_metric_artifact(
+    dataset_name, temp_dir, artifact_name, raw_artifact, custom_metric_index, custom_metric_name
+):
+    if isinstance(raw_artifact, EvaluationArtifact):
+        return raw_artifact
+
+    exception_header = _get_custom_metric_exception_header(custom_metric_index, custom_metric_name)
+
+    # If path is str, convert to pathlib.Path. Then, type inference through file extension
+    if isinstance(raw_artifact, str):
+        raw_artifact = pathlib.Path(raw_artifact)
+    if isinstance(raw_artifact, pathlib.Path):
+        ext_to_artifact_map = {
+            ".png": ImageEvaluationArtifact,
+            ".jpg": ImageEvaluationArtifact,
+            ".jpeg": ImageEvaluationArtifact,
+            ".json": JsonEvaluationArtifact,
+            ".npy": NumpyEvaluationArtifact,
+            ".csv": CsvEvaluationArtifact,
+            ".parquet": ParquetEvaluationArtifact,
+        }
+        if raw_artifact.is_file() and raw_artifact.suffix in ext_to_artifact_map.keys():
+            # copy file to temp location with file name in the defined format
+            artifact_file_name = _gen_log_key(artifact_name, dataset_name) + raw_artifact.suffix
+            artifact_file_local_path = temp_dir.path(artifact_file_name)
+            shutil.copyfile(raw_artifact, artifact_file_local_path)
+
+            mlflow.log_artifact(artifact_file_local_path)
+            return ext_to_artifact_map[raw_artifact.suffix](
+                uri=mlflow.get_artifact_uri(artifact_file_name)
+            )
+        raise MlflowException(
+            f"{exception_header} produced an unsupported artifact '{artifact_name}' with path "
+            f"'{raw_artifact}'. The supported file extensions are:"
+            f".png, .jpg, .jpeg, .json, .npy, .csv, .parquet"
+        )
+
+    import matplotlib.pyplot as pyplot
+
+    # Given object, type inference
+    type_to_ext_map = {
+        pd.DataFrame: ".csv",
+        np.ndarray: ".npy",
+        pyplot.Figure: ".png",
+    }
+
+    type_to_artifact_map = {
+        pd.DataFrame: CsvEvaluationArtifact,
+        np.ndarray: NumpyEvaluationArtifact,
+        pyplot.Figure: ImageEvaluationArtifact,
+    }
+
+    artifact_file_name = _gen_log_key(artifact_name, dataset_name) + type_to_ext_map.get(
+        type(raw_artifact), ".json"
+    )
+
+    artifact_file_local_path = temp_dir.path(artifact_file_name)
+
+    if isinstance(raw_artifact, pd.DataFrame):
+        raw_artifact.to_csv(artifact_file_local_path, index=False)
+    elif isinstance(raw_artifact, np.ndarray):
+        np.save(artifact_file_local_path, raw_artifact, allow_pickle=False)
+    elif isinstance(raw_artifact, pyplot.Figure):
+        raw_artifact.savefig(artifact_file_local_path)
```

Comment:
If we're not permitting `**kwargs` here, we should probably mention in the docs that the only way to modify (globally) any generated plots (dpi, scaling, whitespace bbox, padding, format) during `.savefig()` is through setting values to the env variable dictionary `rcParams`.

Reply:
Good point, I will note this down in the docs. If users want customizations to their figure, they could opt to call `savefig()` first with their desired arguments, then return the file path of their image artifact in their custom metric function.